### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by domoritz**
Clean up types and unused functions.

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -30,13 +30,13 @@ export function reduce(obj, f, init, thisArg?) {
...
 
-export function map(obj, f, thisArg?) {
+export function map(obj, f: (a, d, k, o) => any, thisArg?) {
   if (obj.map) {
     return obj.map.call(thisArg, f);
   } else {
     var output = [];
     for (var k in obj) {
-      output.push( f.call(thisArg, obj[k], k, obj));
+      output.push(f(thisArg, obj[k], k, obj));
```
Comment (kanitw) [inline src/util.ts]: `f.call`  is correct!  -- we bind `this` argument to the function

Comment (domoritz) [inline src/util.ts]: Reverted. The change did not break any tests, though.

Comment (domoritz) [inline src/util.ts]: All these functions are used in compass, right? Or can I delete them?

File: src/compiler/Model.ts [hunk 2]
```diff
@@ -185,20 +185,20 @@ export class Model {
...
   }
 
-  numberFormat = function(channel?: Channel) {
+  numberFormat(channel?: Channel): string {
     // TODO(#497): have different number format based on numberType (discrete/continuous)
     return this.config('numberFormat');
   };
 
-  map(f) {
+  map(f: (...any) => any) {
```
Comment (kanitw) [inline src/compiler/Model.ts]: f should have same signature as vlEncoding.reduce ... same below

**Pull request merged**